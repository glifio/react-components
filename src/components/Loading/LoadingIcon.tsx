import styled, { css, keyframes } from 'styled-components'
import PropTypes from 'prop-types'
import { Colors } from '../theme'

const LoadingIconAnimation = keyframes`
  20% { transform: translate(0, 0) }
  40% { transform: translate(100%, 0) }
  60% { transform: translate(100%, 100%) }
  80% { transform: translate(0, 100%) }
  100% { transform: translate(0, 0) }
`

const LoadingIconEl = styled.span`
  position: relative;
  display: inline-block;
  background-color: ${Colors.PURPLE_MEDIUM};

  ${props => css`
    width: ${props.size};
    height: ${props.size};
    margin: ${props.margin};
  `}

  &:after {
    position: absolute;
    display: block;
    content: '';
    top: 0;
    left: 0;
    background-color: ${Colors.PURPLE_DARK};
    animation-name: ${LoadingIconAnimation};
    animation-duration: 2s;
    animation-iteration-count: infinite;

    ${props => css`
      width: calc(${props.size} * 0.5);
      height: calc(${props.size} * 0.5);
    `}
  }
`

export const LoadingIcon = ({ size, margin }: LoadingIconProps) => (
  <LoadingIconEl size={size} margin={margin} />
)

interface LoadingIconProps {
  size?: string
  margin?: string
}

LoadingIcon.propTypes = {
  size: PropTypes.string,
  margin: PropTypes.string
}

LoadingIcon.defaultProps = {
  size: '3em',
  margin: '0'
}
