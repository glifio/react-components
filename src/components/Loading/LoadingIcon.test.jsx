import { render } from '@testing-library/react'
import { LoadingIcon } from './LoadingIcon'

describe('LoadingIcon', () => {
  test('renders correctly', () => {
    const { container } = render(<LoadingIcon />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
