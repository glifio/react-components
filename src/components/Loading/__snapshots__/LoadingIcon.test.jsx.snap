// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIcon renders correctly 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  background-color: var(--purple-medium);
  width: 3em;
  height: 3em;
  margin: 0;
}

.c0:after {
  position: absolute;
  display: block;
  content: '';
  top: 0;
  left: 0;
  background-color: var(--purple-dark);
  -webkit-animation-name: gxoozo;
  animation-name: gxoozo;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  width: calc(3em * 0.5);
  height: calc(3em * 0.5);
}

<span
  class="c0"
/>
`;
